crates/polars-ops/src/chunked_array/list/namespace.rs

```diff
@@ -255,34 +256,102 @@ pub trait ListNameSpaceImpl: AsList {
         }
     }
 
+    fn eval_lambda_on_amort(
+        s_ref: &Series,
+        lambda_expression: Arc<LambdaExpression>,
+        index_data: Option<&mut (MutablePrimitiveArray<i32>, AmortSeries)>,
+    ) -> PolarsResult<Series> {
+        let s_len = s_ref.len() as i32;
+        (if let Some((index_arr, index_amort)) = index_data {
+            let index_count = index_arr.len() as i32;
+            match index_count.cmp(&s_len) {
+                Ordering::Less => index_arr.extend_trusted_len_values(index_count..s_len),
+                Ordering::Greater => index_arr.truncate(s_len as usize),
+                Ordering::Equal => {} // Do nothing
+            }
+
+            // Create a temporary Box<dyn Array> for this iteration, this is unsafe code but I think should be ok here??
+            let mut temp_box: Box<dyn Array> = unsafe { std::mem::transmute(index_arr.as_box()) };
+
+            unsafe {
```

Comment:
IIUC, this essentially makes the entire function unsafe, since it applies constraints on `index_arr` which is an argument from outside.

Reply:
After removing the transmute the only remaining  unsafe function is with_array, which restores the previous state after swapping the memory, so I don't deem the whole function unsafe